#!/usr/bin/env python
PACKAGE = "slaw_navigation"

#import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("cutoff_angular", double_t, 0, "A double parameter", 0.0, 0.01, 5)
gen.add("KP_angular", double_t, 0, "A double parameter", 0.0, 1.5, 5)
gen.add("KI_angular", double_t, 0, "A double parameter", 0.0, 0, 5)
gen.add("KD_angular", double_t, 0, "A double parameter", 0.0, 1, 5)
gen.add("Ilimit_angular", double_t, 0, "A double parameter", 0.0, 0, 5)
    
gen.add("cutoff_linear_x", double_t, 0, "A double parameter", 0.0, 0.005, 5)
gen.add("KP_linear_x", double_t, 0, "A double parameter", 0.0, 1.5, 5)
gen.add("KI_linear_x", double_t, 0, "A double parameter", 0.0, 0, 5)
gen.add("KD_linear_x", double_t, 0, "A double parameter", 0.0, 1, 5)
gen.add("Ilimit_linear_x", double_t, 0, "A double parameter", 0.0, 0, 5)

gen.add("cutoff_linear_y", double_t, 0, "A double parameter", 0.0, 0.005, 5)
gen.add("KP_linear_y", double_t, 0, "A double parameter", 0.0, 1.5, 5)
gen.add("KI_linear_y", double_t, 0, "A double parameter", 0.0, 0, 5)
gen.add("KD_linear_y", double_t, 0, "A double parameter", 0.0, 1, 5)
gen.add("Ilimit_linear_y", double_t, 0, "A double parameter", 0.0, 0, 5)

exit(gen.generate(PACKAGE, "pid_pose.py", "pid"))
